tiiuae/ghaf
# Add IDS-VM as a defensive networking mechanism

## Description of changes

This PR adds a new VM that enables defensive mechanisms to internal network as well as to outgoing network. This is going to replace the earlier draft pull request PR #146.

- Initially it utilizes mitmproxy toolset to monitor http and https traffic.
- It establishes a web interface for the tool using mitmweb tool. The interface can be launched from the gui-vm by running command 'mitmweb-ui'. The web interface uses ssh-tunnel to securely pass on the interface.
- The chromium-vm launcher is patched so that it does not create warnings or errors due to certificate created by mitmproxy without relaxing web security too much.
- Also snort and tcpdump packages are included to ids-vm.


## Checklist for things done






## Testing

- After boot configure WiFi normally if necessary.
- Open Chromium and browse for a while.
  - Notice the warning prompt about unsupported command-line flag. This is tradeoff between installing CA certificate manually 
  and ignoring certificate warnings entirely.
- Open GUI-VM terminal and run command 'mitmweb-ui' and type in ssh username(ghaf) and password(ghaf) when requested.
- A new browser window opens with web interface of mitmproxy and you can see http(s) flows.
  - Notice that you will see recorded flows of your previous browsing and real time flows.
